cpp/include/cudf/io/parquet.hpp

```diff
@@ -392,6 +392,12 @@ class column_in_metadata {
   std::vector<column_in_metadata> children;
 
  public:
+  /**
+   * @brief Get the children of this column metadata
+   *
+   */
+  std::vector<column_in_metadata>& get_children() { return children; }
```

Comment:
Accessor returning a mutable reference is not much better than direct access. Given the usage, I would recommend replacing this with something like `add_child(column_in_metadata const&)`. The call site would not change too much, and we avoid temporarily having default-initialized columns in the vector.

Reply:
That's exactly what I was thinking but I did it anyway. I have addressed your concern PTAL